### PR TITLE
k8s_cp turbo mode compliance

### LINKED ISSUE #237: k8s_cp does not delay module load
##### SUMMARY

update: k8s_cp does not use the turbo mode at all.

Before `k8s_cp`'s main() is called, all the resources have to be loaded. This has an negative impact of the performance boost that we can get with turbo mode. Turbo mode works in two steps:

- regular: load the module, evaluate the parameters and send the module to the background daemon.
- in the background daemon: load the module and run the main() function.

The second steps should be fast because these dependencies are likely to be already in memory. We don't need most of k8s_cp.py during the first step. This content should be moved in a module_utils file that we lazy-load.

##### ISSUE TYPE
- Bug Report

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

closes #237 

##### ISSUE TYPE

- Feature Pull Request
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/k8s_cp.py [hunk 1]
```diff
@@ -573,32 +192,17 @@ def execute_module(module):
...
-        k8s_copy = load_class.get(state)(module, k8s_ansible_mixin.client)
-        k8s_copy.run()
-    except Exception as e:
-        module.fail_json("Failed to copy object due to: {0}".format(to_native(e)))
+    state = module.params.get("state")
+    if state == "to_pod":
+        k8s_copy = K8SCopyToPod(module, k8s_ansible_mixin.client)
+    else:
+        k8s_copy = K8SCopyFromPod(module, k8s_ansible_mixin.client)
+    k8s_copy.run()
```
Comment (gravesm) [inline plugins/modules/k8s_cp.py:200]: Should we wrap this `run()` call in a try/except block?

**Review by alinabuzachis — COMMENTED** [review]
Am I wrong or the check_mode logic is not handled here (even if the module has supports_check_mode=True)?

**Comment by abikouo** [comment]
> Am I wrong or the check_mode logic is not handled here (even if the module has supports_check_mode=True)?

check_mode is not supported by the module even if it states that, but this should be addressed in another PR
This PR is resolving turbo mode compliance only

**Comment by alinabuzachis** [comment]
> > Am I wrong or the check_mode logic is not handled here (even if the module has supports_check_mode=True)?
> 
> check_mode is not supported by the module even if it states that, but this should be addressed in another PR This PR is resolving turbo mode compliance only

So, I was just wonder if it would be better to have it set to False if it's not supported. @gravesm

**Comment by gravesm** [comment]
> check_mode is not supported by the module even if it states that, but this should be addressed in another PR
This PR is resolving turbo mode compliance only

That's fine. Please, create an issue for it.

File: plugins/modules/k8s_cp.py (hunk 1 above)
Comment (abikouo) [reply plugins/modules/k8s_cp.py:200]: yes indeed, done

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
@alinabuzachis @gravesm the build succeeded and #380 has been created to track support for check_mode.
Please add your final review or comment if something is missing

**Comment by abikouo** [comment]
regate

**Comment by abikouo** [comment]
regate

**Comment by abikouo** [comment]
regate

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
regate

**Pull request merged**